src/UKHO.ERPFacade.Common/HttpClients/EESClient.cs

```diff
@@ -1,20 +1,87 @@
 ﻿using System.Diagnostics.CodeAnalysis;
+using System.Net.Http.Headers;
+using System.Text;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Polly;
+using Polly.Extensions.Http;
+using UKHO.ERPFacade.Common.Authentication;
+using UKHO.ERPFacade.Common.Configuration;
+using UKHO.ERPFacade.Common.Logging;
+using UKHO.ERPFacade.Common.Models;
+using UKHO.ERPFacade.Common.Models.CloudEvents;
 
 namespace UKHO.ERPFacade.Common.HttpClients
 {
     [ExcludeFromCodeCoverage]
     public class EESClient : IEESClient
     {
         private readonly HttpClient _httpClient;
+        private readonly ILogger<EESClient> _logger;
+        private readonly ITokenProvider _tokenProvider;
+        private readonly IOptions<EESConfiguration> _eesConfiguration;
+        private readonly IOptions<RetryPolicyConfiguration> _retryPolicyConfiguration;
+        private readonly IConfiguration _configuration;
 
-        public EESClient(HttpClient httpClient)
+        public EESClient(HttpClient httpClient, ILogger<EESClient> logger, ITokenProvider tokenProvider, IOptions<EESConfiguration> eesConfiguration, IOptions<RetryPolicyConfiguration> retryPolicyConfiguration, IConfiguration configuration)
         {
             _httpClient = httpClient;
+            _logger = logger;
+            _tokenProvider = tokenProvider;
+            _eesConfiguration = eesConfiguration;
+            _retryPolicyConfiguration = retryPolicyConfiguration;
+            _configuration = configuration;
         }
 
         public async Task<HttpResponseMessage> Get(string url)
         {
             return await _httpClient.GetAsync(url);
         }
+
+        public async Task<Result> PostAsync(BaseCloudEvent cloudEvent)
+        {
+            var cloudEventPayload = JsonConvert.SerializeObject(cloudEvent);
+
+            if (!_eesConfiguration.Value.UseLocalResources)
+            {
+                string authToken = _tokenProvider.GetTokenAsync($"{_eesConfiguration.Value.ClientId}/{_eesConfiguration.Value.PublisherScope}").Result;
+                _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", authToken);
+            }
+
+            try
+            {
+                _logger.LogInformation(EventIds.StartingEnterpriseEventServiceEventPublisher.ToEventId(), "Attempting to publish {cloudEventType} event to Enterprise Event Service.", cloudEvent.Type);
+
+                var _retryPolicy = HttpPolicyExtensions
```

Comment:
Add retry policy in Program.cs as PolicyHandler

Reply:
Resolved as per review comment